### PR TITLE
Make morphable entities selectable in UI

### PR DESCRIPTION
**Pull request by daemyung**
Morphable entities can selectable and morphing is grayed out when playing animations.

스크린샷 2022-01-21 오전 11 03 31

### CODE REVIEW

**Review by pixelflinger — CHANGES_REQUESTED** [review]
Is this updating the version of ImGui to a more recent one and is it required?

If so I would like this to be a separate, independent PR.

**Comment by daemyung** [comment]
> Is this updating the version of ImGui to a more recent one and is it required?

Yes it's required. `ImGui::BeginDisabled()` and `ImGui::EndDisabled()` are needed to gray out. But if you want a seperate PR I can do that.

**Comment by daemyung** [comment]
Do not merge until #5083 is merged.

**Review by pixelflinger — COMMENTED** [review]
looks good to me, once it's rebased on top of the imgui update.

File: libs/viewer/include/viewer/SimpleViewer.h
Comment (prideout) [inline libs/viewer/include/viewer/SimpleViewer.h:239]: Personally I kind of like the auto-play feature that we had.  We could keep this set to `1` if we were to change the conditional as I've described in my subsequent comment.

By the way, we should probably have a comment here that explains that it is a 1-based index and 0 means "not playing animation".  That's my fault.

File: libs/viewer/src/SimpleViewer.cpp
Comment (prideout) [inline libs/viewer/src/SimpleViewer.cpp:995]: Instead of testing current animation against 0 in multiple places, you could just say:

```
const bool isAnimation = mCurrentAnimation != && mAnimator->->getAnimationCount() > 0;
```

This would allow us to keep our auto-play feature.

**Review by prideout — CHANGES_REQUESTED** [review]
Looks good, just a comment about auto-play.  Also needs a rebase like Mathias mentioned.

File: libs/viewer/include/viewer/SimpleViewer.h
Comment (daemyung) [reply libs/viewer/include/viewer/SimpleViewer.h]: Ok it's reverted.

File: libs/viewer/src/SimpleViewer.cpp
Comment (daemyung) [reply libs/viewer/src/SimpleViewer.cpp]: Ok it's updated.

**Review by pixelflinger — APPROVED** [review]

**Review by prideout — APPROVED** [review]

**Pull request merged**
